engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java

```diff
@@ -79,10 +83,8 @@ protected void removeDesiredStateChange(StateChange stateChange) {
         desiredStateResets.remove(stateChange.getDefaultInstance());
     }
 
-    protected void refreshTaskList() {
-        if (taskListGenerator != null) {
-            taskListGenerator.requestRefresh();
-        }
+    protected void requestTaskListRefresh() {
+        worldRenderer.requestRenderTaskListRefresh();
```

Comment:
In line with other comment, I recommend to remove this method and have the `worldRenderer.requestRenderTaskListRefresh();` line right in the nodes that need it. Indeed there is some code repetition but it is minimal.